File: data/snippets/quiz016.cpp
class OrderSystem {
private:
    std::string databaseConnectionString;
    bool isLegacyMode;  // 未使用的标志
    int tempVariable;   // 未使用的变量
    
public:
    OrderSystem() {
        databaseConnectionString = "mysql://localhost:3306/orders";
        isLegacyMode = false;  // 这个标志在代码中从未使用
        setupConnection();
    }
    
    void processOrder(int orderId, double amount) {
        validateOrder(orderId);
        updateInventory(orderId);
        sendConfirmation(orderId, amount);
        
        // 永远不会执行的代码
        if (false) {
            legacyProcessing();  // 这个代码分支永远不会执行
        }
    }
    
    // 被其他方法替代，现在不再被调用
    void legacyProcessing() {
        std::cout << "Using legacy order processing..." << std::endl;
        // 过时的处理逻辑
    }
    
    // 被注释掉的功能
    /*
    void oldPaymentMethod() {
        // 这个支付方法已经不再使用，但代码仍保留在这里
        std::cout << "Processing payment with old method" << std::endl;
    }
    */
    
    void generateReport() {
        return;  // 提前返回，下面的代码永远不会执行
        std::cout << "Generating detailed report..." << std::endl;
        calculateStatistics();
    }
    
private:
    void setupConnection() {
        std::cout << "Connecting to database..." << std::endl;
    }
    
    void validateOrder(int orderId) {
        std::cout << "Validating order #" << orderId << std::endl;
    }
    
    void updateInventory(int orderId) {
        std::cout << "Updating inventory for order #" << orderId << std::endl;
    }
    
    void sendConfirmation(int orderId, double amount) {
        std::cout << "Sending confirmation for order #" << orderId << " with amount $" << amount << std::endl;
    }
    
    // 这个私有方法从未被调用
    void calculateStatistics() {
        std::cout << "Calculating order statistics..." << std::endl;
    }
    
    // 另一个未使用的私有方法
    void archiveOldOrders() {
        std::cout << "Archiving old orders..." << std::endl;
    }
};

File: data/snippets/quiz016.java
import java.util.List;
import java.util.ArrayList;

public class OrderSystem {
    private String databaseConnectionString;
    private boolean isLegacyMode;
    private int tempVariable;
    private List<String> unusedList;
    
    public OrderSystem() {
        databaseConnectionString = "mysql://localhost:3306/orders";
        isLegacyMode = false;
        tempVariable = 0;
        unusedList = new ArrayList<>();
        setupConnection();
    }
    
    public void processOrder(int orderId, double amount) {
        validateOrder(orderId);
        updateInventory(orderId);
        sendConfirmation(orderId, amount);
        
        if (false) {
            legacyProcessing();
        }
        
        boolean neverTrue = false;
        if (neverTrue) {
            archiveOldOrders();
        }
    }
    
    public void legacyProcessing() {
        System.out.println("Using legacy order processing...");
    }
    
    private void generateReport() {
        return;
        System.out.println("Generating detailed report...");
        calculateStatistics();
        generateCharts();
    }
    
    /*
    public void processPayment(double amount) {
        if (amount > 0) {
            System.out.println("Processing payment of $" + amount);
            return;
        }
        
        System.out.println("This line will never be reached");
        handleInvalidPayment();
    }
    */
    
    private void setupConnection() {
        System.out.println("Connecting to database...");
    }
    
    private void validateOrder(int orderId) {
        System.out.println("Validating order #" + orderId);
    }
    
    private void updateInventory(int orderId) {
        System.out.println("Updating inventory for order #" + orderId);
    }
    
    private void sendConfirmation(int orderId, double amount) {
        System.out.println("Sending confirmation for order #" + orderId + " with amount $" + amount);
    }
    
    private void calculateStatistics() {
        System.out.println("Calculating order statistics...");
    }
    
    private void archiveOldOrders() {
        System.out.println("Archiving old orders...");
    }
    
    private void generateCharts() {
        System.out.println("Generating charts...");
    }
    
    private void handleInvalidPayment() {
        System.out.println("Handling invalid payment...");
    }
    
    private String formatCurrency(double amount) {
        return "$" + String.format("%.2f", amount);
    }
}
